### PR TITLE
Add display attributes into LocalConnectionStatus

### PR DESCRIPTION
**Pull request by bboreham**
"Display attributes" is some extra data that the overlay wishes to communicate to humans or monitoring systems.

For https://github.com/weaveworks/weave/issues/2663

### CODE REVIEW

File: status.go [hunk 1]
```diff
@@ -174,11 +175,11 @@ func makeLocalConnectionStatusSlice(cm *connectionMaker) []LocalConnectionStatus
 					info = fmt.Sprintf("%-11v %v", "unencrypted", info)
 				}
 			}
-			slice = append(slice, LocalConnectionStatus{conn.remoteTCPAddress(), conn.isOutbound(), state, info})
+			slice = append(slice, LocalConnectionStatus{conn.remoteTCPAddress(), conn.isOutbound(), state, info, lc.OverlayConn.DisplayData()})
```
Comment (brb) [inline status.go]: a dummy race: an overlay has changed after calling `lc.OverlayConn.DisplayName()`.

File: overlay.go [hunk 2]
```diff
@@ -84,6 +84,9 @@ type OverlayConnection interface {
 
 	// DisplayName returns the user-facing overlay name.
 	DisplayName() string
+
+	// DisplayData returns some further data that users may wish to check or monitor
+	DisplayData() map[string]interface{}
```
Comment (brb) [inline overlay.go]: Humble nitpicking: s/`DisplayData`/`Attrs`, as a) `Display` is redundant; b) `Data` is very ambiguous.

File: status.go (hunk 1 above)
Comment (bboreham) [inline status.go]: I guess it's possible, but we also access `conn` 6 times with no locking - should we fix that too?

Comment (brb) [inline status.go]: Not sure about `conn`, as `makeLocalConnectionStatusSlice` is executed by the actor process. (can `conn` be modified outside the actor loop?)

The easy fix is to merge `DisplayName` with `DisplayData` into `GetDisplayNameAndAttrs` (as you suggested in the beginning).

**Comment by awh** [comment]
@bboreham linter is moaning

**Comment by brb** [comment]
LGTM.

**Pull request merged**